src/transformers/modeling_t5.py

```diff
@@ -365,9 +393,16 @@ def unshape(x):
                 k, v = past_key_value_state
 
         if self.is_decoder and use_cache is True:
-            present_key_value_state = ((k, v),)
+            #present_key_value_state = ((k, v),)
+            # HACK -- need to flatted here -- grad checkpoint does not like tuple of tuples
+            if False:
+                print('K, V sizes:')
+                print(k.shape)
+                print(v.shape)
+            present_key_value_state = (torch.stack([k,v], dim=0), )
```

Comment:
Key point -- need to return (k,v) as a Torch object not list -- grad checkpoint requires it.